app/stac_api/models.py

```diff
@@ -125,48 +145,131 @@ class Collection(models.Model):
     # collection
-    summaries_eo_gsd = ArrayField(models.FloatField(), blank=True, null=True)
-    summaries_proj = ArrayField(models.IntegerField(), blank=True, null=True)
-    # after discussion with Chris and Tobias: geoadmin_variant will be an
-    # array field of CharFields. Simple validation is done (e.g. no "Sonderzeichen"
-    # in array)
-    # geoadmin_variant will be also auto-populated on every
-    # update of an asset
-    geoadmin_variant = ArrayField(models.CharField(max_length=15), blank=True, null=True)
-
+    summaries = models.JSONField(
+        default=get_default_summaries_value, encoder=DjangoJSONEncoder, editable=False
+    )
     title = models.CharField(blank=True, max_length=255)  # string
 
     def __str__(self):
         return self.collection_name
 
-    def clean(self):
-        # TODO: move this check to the items save()
-        if self.start_date is None and self.end_date is None:
-            raise ValidationError(_('At least a start date or an end date has to be defined.'))
+    def update_geoadmin_variants(self, asset_geoadmin_variant, asset_proj_epsg, asset_eo_gsd):
+        '''
+        updates the collection's summaries when assets are updated or raises
+        errors when this fails.
+        :param asset_geoadmin_value: asset's value for geoadmin_variant
+        :param asset_proj_epsg: asset's value for proj:epsg
+        :param asset_eo_gsd: asset's value for asset_eo_gsd
+        For all the given parameters this function checks, if the corresponding
+        parameters of the collection need to be updated. If so, they will be either
+        updated or an error will be raised, if updating fails.
+        '''
+        try:
+
+            logger.debug(
+                "updating geoadmin:variants, self.summaries['eo:gsd']=%s, asset_eo_gsd=%s",
+                self.summaries["geoadmin:variant"],
+                asset_eo_gsd
+            )
+
+            if self.summaries["geoadmin:variant"] is None:
+                self.summaries["geoadmin:variant"] = [asset_geoadmin_variant]
+                self.save()
+            elif asset_geoadmin_variant not in self.summaries["geoadmin:variant"]:
+                self.summaries["geoadmin:variant"].append(asset_geoadmin_variant)
+                self.save()
+
+            if self.summaries["proj:epsg"] is None:
+                self.summaries["proj:epsg"] = [asset_proj_epsg]
+                self.save()
+            elif asset_proj_epsg not in self.summaries["proj:epsg"]:
+                self.summaries["proj:epsg"].append(asset_proj_epsg)
+                self.save()
+
+            if self.summaries["eo:gsd"] is None:
+                self.summaries["eo:gsd"] = [asset_eo_gsd]
+                self.save()
+            elif not float_in(asset_eo_gsd, self.summaries["eo:gsd"]):
+                self.summaries["eo:gsd"].append(asset_eo_gsd)
+                self.save()
+
+        except (KeyError, IndexError) as err:
```

Comment:
Remove unnecessary `IndexError` from `try .... except`

Reply:
done

Reply:
this one should stay I guess

Reply:
It should be `except KeyError as err:` as there should not have any `IndexError` in this code, only `KeyError`